RoosterPlanner.Api/Controllers/ShiftController.cs

```diff
@@ -57,6 +57,31 @@ public async Task<ActionResult<List<ShiftViewModel>>> GetShiftsAsync(Guid id)
             }
         }
 
+        [HttpGet("{id}/{userId}/{date}")]
+        public async Task<ActionResult<List<ShiftViewModel>>> GetShiftsAsync(Guid id, Guid userId, DateTime date)
+        {
+            if (id == Guid.Empty)
+                return BadRequest("No valid projectId");
+            if (userId == Guid.Empty)
+                return BadRequest("No valid userI");
+            try
+            {
+                TaskListResult<Shift> result = await shiftService.GetShiftsAsync(id,userId,date);
+                if (!result.Succeeded)
+                    return UnprocessableEntity(new ErrorViewModel {Type = Type.Error, Message = result.Message});
+                if (result.Data == null || result.Data.Count == 0)
+                    return Ok(new List<ShiftViewModel>());
+                List<ShiftViewModel> shiftVmList = result.Data.Select(ShiftViewModel.CreateVm).ToList();
+                return Ok(shiftVmList);
+            }
+            catch (Exception ex)
+            {
+                string message = GetType().Name + "Error in " + nameof(GetShiftAsync);
+                logger.LogError(ex, message);
+                return UnprocessableEntity(new ErrorViewModel {Type = Type.Error, Message = message});
+            }
+        }
+
         [HttpGet("shift/{id}")]
```

Comment:
call url is hier api/shift/shift/{id} ??

Reply:
Aangepast naar api/shift/{id}